internal/signalfx-agent/pkg/core/dpfilters/overridable.go

```diff
@@ -47,3 +49,23 @@ func (f *overridableDatapointFilter) Matches(dp *datapoint.Datapoint) bool {
 	return (f.metricFilter == nil || f.metricFilter.Matches(dp.Metric)) &&
 		(f.dimFilter == nil || f.dimFilter.Matches(dp.Dimensions))
 }
+
+func (f *overridableDatapointFilter) MatchesMetric(m pmetric.Metric) bool {
+	var attributes pcommon.Map
+	switch m.Type() {
+	case pmetric.MetricTypeGauge:
+		attributes = m.Gauge().DataPoints().At(0).Attributes()
+	case pmetric.MetricTypeSummary:
+		attributes = m.Summary().DataPoints().At(0).Attributes()
+	case pmetric.MetricTypeSum:
+		attributes = m.Sum().DataPoints().At(0).Attributes()
+	case pmetric.MetricTypeHistogram:
+		attributes = m.Histogram().DataPoints().At(0).Attributes()
+	case pmetric.MetricTypeExponentialHistogram:
+		attributes = m.ExponentialHistogram().DataPoints().At(0).Attributes()
```

Comment:
Is this expected that we drop the whole metric only if the first data point matches? It doesn't look like the case for the existing behavior and probably should not be changed this way. 

The filtering logic is probably a bit more complicated. We cannot just pass `pmetric.Metric` and return a `bool`. We need to be able to mutate the `pmetric.Metric` and drop some datapoints.

I'd suggest we merge this change without filtering logic under a disabled feature gate first. Then we handle the filtering and enable the feature gate. WDYT?

cc @rmfitzpatrick

Reply:
Good call that the implementations can't be 1:1. I'm ok w/ adding the deeper filtering here or tabling until a later pass w/ clear disclaimer in release notes. Either way the feature should be gated, though my preference is for a monitor config update that's false by default to avoid a global config mechanism but can see it either way.

Reply:
If we table it, my preference would be not to introduce any support for filtering config under the feature gate (we can fail to start if this config is used). As the next step, we introduce the correct filtering, then enable the feature gate. This filtering solution doesn't seem right to give it to users even under a feature gate.

Reply:
To clarify I was referring to adding the config option and removing the filtering in this PR (maybe failing out if filter fields are used like you suggest) and then adding filtering support in a later one. I wouldn't suggest using the current implementation w/ its known issues.